ZohoCRMCharts/Classes/ZCRMKPI.swift

```diff
@@ -0,0 +1,587 @@
+	*/
+	private func renderKPIView() {
+		
+		self.setUpTitleView()
+		self.updateTitleText()
+		if self.isRankings || self.isScorecard {
+			self.tableView.dataSource = self
+			self.tableView.delegate = self
+			self.setKpiView()
+		} else {
+			self.setSimpleKpiView()
+		}
+	}
+	
+	/**
+		Sets the KPI data to the views.
+	*/
+	fileprivate func renderKPIData() {
+		
+		if self.isRankings || self.isScorecard {
+			self.setKpiData()
+		} else {
+			self.setSimpleKpiData()
+		}
+	}
+	
+	fileprivate func addConstraints() {
+		
+		self.invalidateIntrinsicContentSize()
+		if self.isRankings || self.isScorecard {
+			self.addKpiConstraints()
+		} else {
+			self.addSimpleKpiConstraints()
+		}
+	}
+	
+	/**
+		Sets the view position of the title label. It will be common for all types.
+	*/
+	private func setUpTitleView() {
+		
+		self.titleView.translatesAutoresizingMaskIntoConstraints = false
+		self.addSubview(self.titleView)
+		var constraints: [NSLayoutConstraint] = []
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "H:|-15-[title]", options: [], metrics: nil, views: ["title": self.titleView])
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:|-15-[title]", options: [], metrics: nil, views: ["title": self.titleView])
+		NSLayoutConstraint.activate(constraints)
+		
+	}
+	
+	/**
+		Sets the view for standard, growth index and basic KPI.
+	*/
+	private func setSimpleKpiView(){
+		
+		self.valueView.translatesAutoresizingMaskIntoConstraints = false
+		self.addSubview(self.valueView)
+		if (self.type != .basic){
+			self.comparedToView.translatesAutoresizingMaskIntoConstraints = false
+			self.comparedToView.textAlignment = .right
+			self.addSubview(self.comparedToView)
+		}
+		
+	}
+	
+	/**
+		Adds the constraints for the simple KPI.
+	*/
+	private func addSimpleKpiConstraints() {
+		
+		var constraints: [NSLayoutConstraint] = []
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:[title(<=25)]-[value(>=title)]", options: [], metrics: nil, views: ["title": self.titleView, "value": self.valueView])
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "H:|-15-[value]", options: [], metrics: nil, views: ["value": self.valueView])
+		if !self.isBasic {
+			constraints += NSLayoutConstraint.constraints(withVisualFormat: "H:[comparedTo]-15-|", options: [.alignAllBottom], metrics: nil, views: ["value": self.valueView, "comparedTo": self.comparedToView])
+			constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:[comparedTo]-|", options: [], metrics: nil, views: ["comparedTo": self.comparedToView])
+		}
+		NSLayoutConstraint.activate(constraints)
+	}
+	
+	/**
+		Sets the data for standard, growth index and basic KPI.
+	*/
+	private func setSimpleKpiData() {
+		
+		let data: ZCRMKPIRow =  self.data[0]
+		if !self.isBasic {
+			self.updateComparedToViewText(text: data.comparedToLabel + ": " + data.comparedToValue)
+		}
+		if self.isStandard {
```

Comment:
use switch case

Reply:
https://github.com/zoho/CRM-iOS-Charts/pull/3#discussion_r248934137

Reply:
How to use switch over there?. Here we are checking a flag, not a value.